tests/unit/test_response/test_response_to_asgi_response.py

```diff
@@ -299,13 +299,21 @@ def handler() -> Template:
             background=background_task,
         )
 
+    app = Litestar(
+        route_handlers=[],
+        template_config=TemplateConfig(
+            engine=JinjaTemplateEngine,
+            directory=tmp_path,
+        ),
+    )
+
     with create_test_client(
         handler, template_config=TemplateConfig(engine=JinjaTemplateEngine, directory=tmp_path)
     ) as client:
         route: HTTPRoute = client.app.routes[0]
         route_handler = route.route_handlers[0]
         response = await route_handler.to_response(
-            data=route_handler.fn.value(), app=client.app, request=RequestFactory().get()
+            data=route_handler.fn.value(), app=client.app, request=RequestFactory(app=app).get()
```

Comment:
Why was this change necessary?

Reply:
The `to_asgi_response`  in [this function](https://github.com/litestar-org/litestar/blob/44687e5b8aedb05002c79c46420fd5c00be7442c/litestar/response/template.py#L89) checks if the app template engine has been configured and raises an error otherwise. This test was raising that configuration error because the `RequestFactory` creates a request with a vanilla app that hasn't had a template engine configured. For that reason I am creating an app instance in this test and passing it through to `RequestFactory` so that the configuration error doesn't get raised and the test behaves like it did before.